docs/middleware.md

```diff
@@ -0,0 +1,34 @@
+# Redis Output Cache Middleware
```

Comment:
This file should have a more meaningful name - "middleware" is more of an implementation detail